### PR TITLE
Features/1203099733411537 add weights after benchmarking

### PR DESCRIPTION
**Pull request by sayyed-kamran**
Benchmarking functions and test setup added. More ever, weights functions are also added.

### CODE REVIEW

File: src/mock.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+// Configure a mock runtime to test the pallet.
+frame_support::construct_runtime!(
+    pub enum Test where
+        Block = Block,
+        NodeBlock = Block,
+        UncheckedExtrinsic = UncheckedExtrinsic,
+    {
+        System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+        Timestamp: pallet_timestamp::{Pallet, Call, Storage, Inherent},
```
Comment (sfffaaa) [inline src/mock.rs]: Does the storage pallet have a dependency on the Timestamp?

Comment (sayyed-kamran) [reply src/mock.rs]: I do not think so. Let me work it out.

Comment (sfffaaa) [reply src/mock.rs]: Then please remove all the Timestamp-related stuff in the mock file and Cargo.toml.

Comment (sayyed-kamran) [reply src/mock.rs]: Done

**Review by sfffaaa — APPROVED** [review]

**Pull request merged**